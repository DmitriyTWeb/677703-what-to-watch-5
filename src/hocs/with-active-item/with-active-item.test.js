import React from "react";
import renderer from "react-test-renderer";
import withActiveItem from "./with-active-item";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';

const MockComponent = () => {
  return (
    <div>
    </div>
  );
};

const mockStore = configureStore([]);
const MockComponentWrapped = withActiveItem(MockComponent);

describe(`Test withActiveItem with internal Component connected to Redux`, () => {
  let store;
  let tree;

  beforeEach(() => {
    store = mockStore({
      withActiveItemId: -1,
    });
    store.dispatch = jest.fn();

    tree = renderer.create(
        <Provider store={store}>
          <MockComponentWrapped />
        </Provider>
    );
  });

  it(`withActiveItem is rendered correctly`, () => {
    expect(tree).toMatchSnapshot();
  });
});
