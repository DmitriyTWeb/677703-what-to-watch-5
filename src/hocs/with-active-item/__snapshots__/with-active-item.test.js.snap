// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test withActiveItem with internal Component connected to Redux withActiveItem is rendered correctly 1`] = `<div />`;
